### DISCUSSION TITLE
OTBR cannot handle MLE message when LinkMarginLow occurs

### DISCUSSION THREAD

**Discussion by xiaojiang2017** [openthread/openthread#11020] [General]
**Describe the bug** A clear and concise description of what the bug is.
In our Thread network, we use Thread devices with different solutions. The GW is an OTBR that uses the silicon labs EFR32mg21 solution.
During the test, the log `Failed to process Advertisment: LinkMarginLow` was captured on OTBR.
OTBR is partitioned from other routers and can no longer process MLE messages. Some commands on OTBR cannot be performed, such as the `leaderweight & state` command to switch roles to leader. No workaround has been found so far, except restarting `otbr-agent`.
![1e5a5336cdfb012b27c783f1bbfd888](https://github.com/user-attachments/assets/0f92f3e3-8808-45cb-999c-564344e5d1f7)
We added some debugging code as follows.
```
Error MleRouter::HandleAdvertisement(RxInfo &aRxInfo, uint16_t aSourceAddress, const LeaderData &aLeaderData)
{
    ......
    if (aLeaderData.GetPartitionId() != mLeaderData.GetPartitionId())
    {
        LogWarn("Different partition (peer:%lu, local:%lu)", ToUlong(aLeaderData.GetPartitionId()),
                ToUlong(mLeaderData.GetPartitionId()));

        LogWarn("GL-DEBUG receive Advertisement from 0x%04x", aSourceAddress);
        LogWarn("GL-DEBUG linkMargin: %d = rssi=%d - floor=%d", linkMargin,
                aRxInfo.mMessageInfo.GetThreadLinkInfo()->GetRss(), Get<Mac::Mac>().GetNoiseFloor());

        if (linkMargin < kPartitionMergeMinMargin)
        {
            LogWarn("GL-DEBUG kErrorLinkMarginLow");
        }
```
![9831a0f496c0736735d304ee7d6608c](https://github.com/user-attachments/assets/d7816685-be2c-49e1-9e39-1c13f276f1b7)

**To Reproduce** Information to reproduce the behavior, including:

1. Git commit id
2. IEEE 802.15.4 hardware platform **Silicon Labs GSDK v4.4.3**
3. Build steps
4. Network topology

**Expected behavior** A clear and concise description of what you expected to happen.

**Console/log output** If applicable, add console/log output to help explain your problem.

**Additional context** Add any other context about the problem here.

**Comment by xiaojiang2017** [comment]
The router number and its rloc16 in the picture are,
```
13 --> bc00
14 --> 3400
7  --> 5c00
8  --> 8000
10 --> c400
```

**Comment by jwhui** [comment]
Wireless communication properties are time varying due to changes in environment. To be robust to time-varying conditions, OpenThread requires a minimum link margin to establish links:

https://github.com/openthread/openthread/blob/d540fc7f43b41da8134fb42c76dc8aeb84b6c563/src/core/config/mle.h#L213

While it may be possible to establish a link with low link margin at a given point in time, such links are more likely to become unusable in the future.

**Reply by xiaojiang2017** [reply]
Yeah. But the OTBR will enter an abnormal state in our test environment and no longer send Advertisement packets unless the `otbr-agent` is restarted.

**Reply by jwhui** [reply]
Do you observe the radio transmitting any packets? It may be that the radio platform is not properly servicing requests to transmit packets.

**Reply by xiaojiang2017** [reply]
The TBR can perform IEEE 802.15.4 active scans, but it cannot actively send Advertisement packets anymore.
![920cbf61586dbdb9b1e4cf786d84979](https://github.com/user-attachments/assets/509f7eac-489d-4a1c-b6c4-25ff2efe0e38)
